src/allmydata/util/i2p_provider.py

```diff
@@ -148,13 +154,25 @@ def get_i2p_handler(self):
         launch = self._get_i2p_config("launch", False, boolean=True)
         configdir = self._get_i2p_config("i2p.configdir", None)
         keyfile = self._get_i2p_config("dest.private_key_file", None)
+        options = self._get_i2p_config("i2cp.options", None)
+
+        if options:
+            def transform_escapes(val):
+                return val.replace('\,', '\ESCAPED-COMMA').replace('\:', '\ESCAPED-COLON')
+
+            def unescape(val):
+                return val.replace('\ESCAPED-COMMA', ',').replace('\ESCAPED-COLON', ':')
+
```

Comment:
yeah, especially because the existing i2p_provider puts backslashes into `escaped_sam_port`. Maybe we could have a few examples of what the original values of sam_port and i2cp.options are, and what gets written into the config file? And some clear rules about what is legal in each kind of string and what is not.